src/components/MAPI/organizations/OrganizationListCreateOrg.tsx

```diff
@@ -0,0 +1,179 @@
+import { useHttpClient } from 'lib/hooks/useHttpClient';
+import * as securityv1alpha1 from 'model/services/mapi/securityv1alpha1';
+import PropTypes from 'prop-types';
+import React, { useLayoutEffect, useMemo, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { organizationsLoadMAPI } from 'stores/organization/actions';
+import Button from 'UI/Controls/Button';
+import TextInput from 'UI/Inputs/TextInput';
+
+import {
+  extractErrorMessage,
+  OrganizationNameStatusMessage,
+  validateOrganizationName,
+} from './utils';
+
+const VALIDATION_ERROR_DEBOUNCE_RATE = 500;
+
+interface IOrganizationListCreateOrgProps
+  extends React.ComponentPropsWithoutRef<typeof Collapsible> {
+  onSubmit?: () => void;
+  onCancel?: () => void;
+}
+
+const OrganizationListCreateOrg: React.FC<IOrganizationListCreateOrgProps> = ({
+  onSubmit,
+  onCancel,
+  open,
+  ...props
+}) => {
+  const [orgName, setOrgName] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+
+  const [hasTyped, setHasTyped] = useState(false);
+
+  const validationMessage = useMemo(() => {
+    if (!hasTyped) return OrganizationNameStatusMessage.Ok;
+
+    return validateOrganizationName(orgName).statusMessage;
+  }, [hasTyped, orgName]);
+
+  const debouncedValidationMessage = useDebounce(
+    validationMessage,
+    VALIDATION_ERROR_DEBOUNCE_RATE
+  );
+  const isValid = validationMessage.length < 1;
+
+  const resetForm = () => {
+    setHasTyped(false);
+    setIsCreating(false);
+    setOrgName('');
+  };
+
+  const onChangeOrgName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!hasTyped) {
+      setHasTyped(true);
+    }
+
+    setOrgName(e.target.value);
+  };
+
+  const handleCancel = () => {
+    onCancel?.();
+
+    setTimeout(() => {
+      resetForm();
+      // eslint-disable-next-line no-magic-numbers
+    }, 200);
+  };
+
+  const client = useHttpClient();
+  const auth = useAuthProvider();
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (e: React.FormEvent<HTMLElement>) => {
+    e.preventDefault();
+
+    if (!isValid) return;
+
+    setIsCreating(true);
+
+    try {
+      await securityv1alpha1.createOrganization(client, auth, orgName);
+      dispatch(organizationsLoadMAPI(auth));
+
+      onSubmit?.();
+
+      new FlashMessage(
+        `Organization <code>${orgName}</code> created successfully`,
+        messageType.SUCCESS,
+        messageTTL.SHORT
+      );
+
+      setTimeout(() => {
+        resetForm();
+        // eslint-disable-next-line no-magic-numbers
+      }, 200);
+    } catch (err) {
+      onCancel?.();
+
+      new FlashMessage(
+        `Unable to create organization <code>${orgName}</code>`,
+        messageType.ERROR,
+        messageTTL.LONG,
+        extractErrorMessage(err)
+      );
+
+      ErrorReporter.getInstance().notify(err);
+    }
+  };
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useLayoutEffect(() => {
```

Comment:
Focusing the input when the form is opened. This is pretty ugly, but using the `autoFocus` prop does not work, for some reason